llmcode/persistence_layer.py

```
@@ -0,0 +1,1145 @@
+        self.storage_dir.mkdir(exist_ok=True)
+        self.lock = threading.Lock()
+        
+        # Create subdirectories
+        (self.storage_dir / "metadata").mkdir(exist_ok=True)
+        (self.storage_dir / "results").mkdir(exist_ok=True)
+        (self.storage_dir / "indexes").mkdir(exist_ok=True)
+    
+    def _get_cache_file_path(self, cache_key: str, file_type: str) -> Path:
+        """Get file path for cache key and file type"""
+        # Use first 2 characters of hash for subdirectory structure
+        hash_part = hashlib.md5(cache_key.encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)